picard/script/functions.py

```diff
@@ -1381,3 +1381,92 @@ def func_countryname(parser, country_code, translate=""):
     if translate:
         return gettext_countries(name)
     return name
+
+
+DateTuple = namedtuple('DateTuple', ('year', 'month', 'day'))
+
+
+def _split_date(date_to_parse, date_order="ymd"):
+    """Split the specified date into parts.
+
+    Args:
+        date_to_parse (str): Date string to parse
+        date_order (str, optional): Order of date elements. Can be "ymd", "mdy" or "dmy". Defaults to "ymd".
+
+    Returns:
+        tuple: Tuple of the date parts as (year, month, day)
+    """
+
+    parts = re.split('[^0-9]+', date_to_parse.strip())
```

Comment:
Just a minor thing, we can use `\D` (=any non-digit) instead of `[^0-9]`:

```suggestion
    parts = re.split(r'\D+', date_to_parse.strip())
```